File: CPP [1001-1500]/1154.day_of_the_year [1].cpp
class Solution {
public:
    bool isLeapYear(int year){
        return year % 100 == 0 ? year % 400 == 0 : year % 4 == 0;
    }

    int dayOfYear(string date) {
        int yyyy = stoi(date.substr(0, 4));
        int mm = stoi(date.substr(5, 2));
        int dd = stoi(date.substr(8));
        
        vector<int> days = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
        if(isLeapYear(yyyy))
            days[1]++;
            
        int ans = 0;
        for(int i=0; i<mm-1; i++)
            ans += days[i];
        return ans + dd;
    }
};

File: JS/1154.day_of_the_year [1].js
/**
 * @param {string} date
 * @return {number}
 */
var dayOfYear = function(date) {
    const d = new Date(date);
    const yyyy = d.getFullYear();
    const mm = d.getMonth() + 1; // +1 is to convert 0-based to 1-based indexing
    const dd = d.getDate();

    const days = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
    if(new Date(yyyy, 1, 29).getDate() === 29) // leap year
        days[1]++;
    
    let ans = 0;
    for(let i=0; i<mm-1; i++)
        ans += days[i];
    return ans + dd;
};